crates/ruma-events/src/room/redaction.rs

```diff
@@ -457,40 +457,30 @@ impl CanBeEmpty for RoomRedactionUnsigned {
     }
 }
 
-/// Returns the value of the proper `redacts` field for the given room version.
+/// Returns the value of the proper `redacts` field for the given redaction rules.
 ///
-/// If the `redacts` field is not the proper one for the given room version, this falls back to
-/// the one that is available.
+/// If the `redacts` field is not the proper one for the given rules, this falls back to the one
+/// that is available.
 ///
 /// # Panics
 ///
 /// Panics if both `redacts` and `content_redacts` are `None`.
 fn redacts<'a>(
-    room_version: &'_ RoomVersionId,
+    rules: &'_ RedactionRules,
     redacts: Option<&'a EventId>,
     content_redacts: Option<&'a EventId>,
 ) -> &'a EventId {
-    match room_version {
-            RoomVersionId::V1
-            | RoomVersionId::V2
-            | RoomVersionId::V3
-            | RoomVersionId::V4
-            | RoomVersionId::V5
-            | RoomVersionId::V6
-            | RoomVersionId::V7
-            | RoomVersionId::V8
-            | RoomVersionId::V9
-            | RoomVersionId::V10 => redacts
-                .or_else(|| {
-                    error!("Redacts field at event level not available, falling back to the one inside content");
-                    content_redacts
-        })
-                .expect("At least one redacts field is set"),
-            _ => content_redacts
-                .or_else(|| {
-                    error!("Redacts field inside content not available, falling back to the one at the event level");
-                    redacts
-        })
-                .expect("At least one redacts field is set"),
-        }
+    if rules.keep_room_redaction_redacts {
+        content_redacts
+            .or_else(|| {
+                error!("Redacts field inside content not available, falling back to the one at the event level");
+                redacts
+    })
```

Comment:
```suggestion
        })
```

Reply:
I split the string so that rustfmt manages to work here.